File: src/state_handlers/ask_make_cookbook.js
/**
 * @fileOverview
 * @author Peter Mitrano- mitrnanopeter@gmail.com
 */

'use strict';

var Core = require('../core');
var Alexa = require('../alexa');

/**
 * Represents the reponses when we expect the user to request to make a new recipe
 */
module.exports = Alexa.CreateStateHandler(Core.states.ASK_MAKE_COOKBOOK, {
  'AMAZON.YesIntent': function () {
    this.handler.state = Core.states.INGREDIENTS_OR_INSTRUCTIONS;
    this.emit(":ask", "Do you want to start with the ingredients or the instructions?");
  },
  'AMAZON.NoIntent': function () {
    // return to initial state!
    this.handler.state = Core.states.ASK_SEARCH;
    this.emit(":ask", "I could search for one online?");
  },
  'Unhandled': function () {
    this.emit(":ask", "I'm confused. Do you want to make the recipe from your cookbook?");
  }
});

File: my_cookbook/skill/handlers/ask_make_cookbook.py
from my_cookbook.util import core
from my_cookbook.util import responder


class AskMakeCookbookHandler():
    def AMAZON_YesIntent(self, handlers, persistant_attributes, attributes, slots):
        attributes[core.STATE_KEY] = core.States.INGREDIENTS_OR_INSTRUCTIONS
        return responder.ask("Do you want to start with the ingredients or the instructions?", None,
                             attributes)

    def AMAZON_NoIntent(self, handlers, persistant_attributes, attributes, slots):
        attributes[core.STATE_KEY] = core.States.ASK_SEARCH
        return responder.ask("I could search for one online?", None, attributes)

    def Unhandled(self, handlers, persistant_attributes, attributes, slots):
        return responder.ask("I'm confused, Do you want to make the recipe from your cookbook?",
                             None, attributes)


handler = AskMakeCookbookHandler
state = core.States.ASK_MAKE_COOKBOOK
